lib/internal/dns/promises.js

```diff
@@ -154,6 +154,20 @@ function createLookupPromise(family, hostname, all, hints, verbatim) {
 }
 
 const validFamilies = [0, 4, 6];
+/**
+ * @typedef DNSLookupResult = { address: string, family: number }
+ */
+/**
+ * Get the IP address for a given hostname.
+ * @param {URL['hostname']} hostname - The hostname to resolve (ex. 'nodejs.org').
+ * @param {Object} [options] - Optional settings.
+ * @param {boolean} [options.all=false] - Whether to return all or just the first resolved address.
+ * @param {number} [options.family=0] - The record family. Must be 4, 6, or 0 (for both).
+ * @param {number} [options.hints] - One or more supported getaddrinfo flags (supply multiple via bitwise OR).
+ * @param {boolean} [options.verbatim=false] - Return results in same order DNS resolved them;
+ * otherwise IPv4 then IPv6. New code should supply `true`.
```

Comment:
```suggestion
 * @param {LookupOptions} [options] - Optional settings.
 * @typdef {object} LookupOptions
 * @property {boolean} [all] - Whether to return all or just the first resolved address.
 * @property {4 | 6 | 0} [family] - The record family. Must be 4, 6, or 0 (for both).
 * @property {number} [hints] - One or more supported getaddrinfo flags (supply multiple via bitwise OR).
 * @property {boolean} [verbatim = false] - Return results in same order DNS resolved them;
 * otherwise IPv4 then IPv6. New code should supply `true`.
```